code/modules/fishing/fish/fish_types.dm

```diff
@@ -0,0 +1,242 @@
+	stable_population = 2
+	fillet_type = /obj/item/food/fishmeat/moonfish
+	average_size = 100
+	average_weight = 2000
+
+/obj/item/fish/gunner_jellyfish
+	name = "gunner jellyfish"
+	desc = "So called due to their resemblance to an artillery shell, the gunner jellyfish is native to Tizira, where it is enjoyed as a delicacy. Produces a mild hallucinogen that is destroyed by cooking."
+	icon_state = "gunner_jellyfish"
+	required_fluid_type = AQUARIUM_FLUID_SALTWATER
+	stable_population = 4
+	fillet_type = /obj/item/food/fishmeat/gunner_jellyfish
+
+/obj/item/fish/needlefish
+	name = "needlefish"
+	desc = "A tiny, transparent fish which resides in large schools in the oceans of Tizira. A common food for other, larger fish."
+	icon_state = "needlefish"
+	required_fluid_type = AQUARIUM_FLUID_SALTWATER
+	stable_population = 12
+	fillet_type = null
+	average_size = 30
+	average_weight = 300
+	fishing_traits = list(/datum/fishing_trait/carnivore)
+
+/obj/item/fish/armorfish
+	name = "armorfish"
+	desc = "A small shellfish native to Tizira's oceans, known for its exceptionally hard shell. Consumed similarly to prawns."
+	icon_state = "armorfish"
+	required_fluid_type = AQUARIUM_FLUID_SALTWATER
+	stable_population = 10
+	fillet_type = /obj/item/food/fishmeat/armorfish
+	fish_ai_type = FISH_AI_SLOW
+
+/obj/item/storage/box/fish_debug
+	name = "box full of fish"
+
+/obj/item/storage/box/fish_debug/PopulateContents()
+	for(var/fish_type in subtypesof(/obj/item/fish))
+		new fish_type(src)
+
+/obj/item/fish/donkfish
+	name = "donk co. company patent donkfish"
+	desc = "A lab-grown donkfish. Its invention was an accident for the most part, as it was intended to be consumed in donk pockets. Unfortunately, it tastes horrible, so it has now become a pseudo-mascot."
+	icon_state = "donkfish"
+	random_case_rarity = FISH_RARITY_VERY_RARE
+	required_fluid_type = AQUARIUM_FLUID_FRESHWATER
+	stable_population = 4
+	fillet_type = /obj/item/food/fishmeat/donkfish
+
+/obj/item/fish/emulsijack
+	name = "toxic emulsijack"
+	desc = "Ah, the terrifying emulsijack. Created in a laboratory, this slimey, scaleless fish emits an invisible toxin that emulsifies other fish for it to feed on. Its only real use is for completely ruining a tank."
+	icon_state = "emulsijack"
+	random_case_rarity = FISH_RARITY_GOOD_LUCK_FINDING_THIS
+	required_fluid_type = AQUARIUM_FLUID_ANADROMOUS
+	stable_population = 3
+
+/obj/item/fish/emulsijack/process(delta_time = SSOBJ_DT)
+	var/emulsified = FALSE
+	var/obj/structure/aquarium/aquarium = loc
+	if(istype(aquarium))
+		for(var/obj/item/fish/victim in aquarium)
+			if(istype(victim, /obj/item/fish/emulsijack))
+				continue //no team killing
+			victim.adjust_health((victim.health - 3) * delta_time) //the victim may heal a bit but this will quickly kill
+			emulsified = TRUE
+	if(emulsified)
+		adjust_health((health + 3) * delta_time)
+		last_feeding = world.time //emulsijack feeds on the emulsion!
+	..()
```

Comment:
don't call parent here if there isn't an /obj/item/fish/process, because the base process proc returns PROCESS_KILL for obvious reasons.

Reply:
There is fish process though.